compiler/src/main/kotlin/io/spine/protodata/Pipeline.kt

```diff
@@ -31,70 +31,49 @@ import io.spine.base.Production
 import io.spine.logging.Logging
 import io.spine.protodata.renderer.Renderer
 import io.spine.protodata.renderer.SourceSet
-import io.spine.protodata.subscriber.CodeEnhancement
-import io.spine.protodata.subscriber.SkipEverything
-import io.spine.protodata.subscriber.Subscriber
-import io.spine.protodata.transport.PrunableTransport
-import io.spine.server.BoundedContext
 import io.spine.server.ServerEnvironment
-import io.spine.server.projection.ProjectionRepository
 import io.spine.server.storage.memory.InMemoryStorageFactory
 
 /**
  * A pipeline which processes the Protobuf files.
  *
- * A pipeline consists of several [Subscriber]s and a single [Renderer] and runs on a single
- * source set.
+ * A pipeline consists of the `ProtoData` context, which receives Protobuf compiler events, and
+ * a single [Renderer]. A pipeline runs on a single source set.
  *
- * The pipeline starts by building a Bounded Context with the supplied subscribers.
- * Then, the Protobuf compiler events are emitted for the subscribers to listen. Subscribers produce
- * [CodeEnhancement]s in response to the events.
- * Then, the [Renderer], based on the generated enhancements, alters the source set. This may
- * include creating new files and/or modifying existing ones.
- * Lastly, the source set is stored onto the file system.
+ * The pipeline starts by building the `ProtoData` bounded context with the supplied
+ * [ContextExtension]s. Then, the Protobuf compiler events are emitted and the subscribers in
+ * the context receive them. Then, the [Renderer], which is able to query the states of entities in
+ * the `ProtoData` context, alters the source set. This may include creating new files and/or
+ * modifying existing ones. Lastly, the source set is stored onto the file system.
  */
 public class Pipeline(
-    private val subscribers: List<Subscriber<*>>,
-    private val projections: List<ProjectionRepository<*, *, *>>,
-    private val renderer: (List<CodeEnhancement>) -> Renderer,
+    private val extensions: List<ContextExtension>,
+    private val renderer:  Renderer,
     private val sourceSet: SourceSet,
     private val request: CodeGeneratorRequest
-) : Logging, AutoCloseable {
-
-    private var generatorContext: BoundedContext? = null
+) : Logging {
 
     init {
         val config = ServerEnvironment.`when`(Production::class.java)
-        config.use(PrunableTransport)
         config.use(InMemoryStorageFactory.newInstance())
     }
 
     /**
      * Executes the processing pipeline.
      */
     public operator fun invoke() {
```

Comment:
Again, I'm not sure we need to add `public` here and above which is default in Kotlin.

Reply:
As discussed, we use the explicit API mode and always mark `public` stuff `public`.